packages/message-core/src/AvMessage.js

```diff
@@ -46,16 +47,28 @@ class AvMessage {
       event = (data && data.event) || this.DEFAULT_EVENT;
     }
 
-    this.onMessage(event, data);
+    const metadata = { isSameWindow };
+
+    this.onMessage(event, data, metadata);
   };
 
-  subscribe(event, fn) {
+  #lastId = 0;
+
+  subscribe(event, callback, options) {
     if (!this.subscribers[event]) {
       this.subscribers[event] = [];
     }
-    this.subscribers[event].push(fn);
+
+    this.#lastId += 1;
+    const id = this.#lastId;
+
+    const ignoreSameWindow = options?.ignoreSameWindow === undefined ? true : options.ignoreSameWindow;
```

Comment:
Double check me, but I think we can shorten this to
`const ignoreSameWindow = options?.ignoreSameWindow ?? true`
If the value is undefined or null, then default to true

Reply:
Good call. Change has been made.

Reply:
I think the change for this line got left of the commit

Reply:
Well I do declare! Fixed. Or is it? 🎶 _suspenseful music_ 🎶